apps/music/js/ui/views/player_view.js

```diff
@@ -244,22 +244,19 @@ var PlayerView = {
   },
 
   setCoverImage: function pv_setCoverImage(fileinfo) {
-    // Reset the image to be ready for fade-in
-    this.offscreenImage.src = '';
-    this.coverImage.classList.remove('fadeIn');
-
-    AlbumArtCache.getCoverURL(fileinfo).then(function(url) {
-      this.offscreenImage.addEventListener('load', pv_showImage.bind(this));
-      this.offscreenImage.src = url;
+    AlbumArtCache.getFullSizeURL(fileinfo).then(function(url) {
+      if (this.offscreenImage.src !== url) {
+        this.coverImage.style.backgroundImage = 'url(' + url + ')';
+        this.coverImage.classList.remove('fadeIn');
+
+        this.offscreenImage.src = url;
+        this.offscreenImage.addEventListener('load', function pv_loaded(event) {
+          /* jshint validthis:true */
+          event.target.removeEventListener('load', pv_loaded);
+          this.coverImage.classList.add('fadeIn');
+        });
+      }
```

Comment:
I think that using hte offscreenImage here is unnecessary and possibly inefficient. 

1)  offscreenImage has never been assigned a size, and the onload handler does not cause it to be rendered anywhere, so I think this means that it will be decoded at full size. Gecko will assume you're trying to do an image cache and will decode for you so it is ready when you insert the image into the document.  I'm not sure how you can test what size the image gets decoded at, however. The patch for this landed late last week and then was backed out because it was not working correctly for code that used background-image.

2) Gecko fires the load event well before the image is actually decoded and ready for display. The onload hander is apparently used to know when to trigger the fadeIn animation.  Since there is no network latency here, the decode time is probably longer than the load time.  I suspect that you could just get rid of the onload handler completely and maybe just add a short animation-delay to the fadeIn animation to compensate.

3) I like that you're're adding the test so that you don't change the image if not necessary.  But you could do the same check on coverImage.style.backgroundImage as long as you add the "url()" before comparing.

4) I don't get how the this inside the event handler could possibly be valid.  Did you mean to add a .bind(this)? Doesn't this refer to the offscreenImage at that point in the code?